pkg/action/action_handler.go

```diff
@@ -142,45 +153,51 @@ func (h *ActionHandler) ExecuteAction(actionExecutionDTO *proto.ActionExecutionD
 }
 
 func (h *ActionHandler) execute(actionItem *proto.ActionItemDTO) error {
+	clusterAPIAction := false
+	if (*actionItem.ActionType == proto.ActionItemDTO_PROVISION ||
+		*actionItem.ActionType == proto.ActionItemDTO_SUSPEND) &&
+		*actionItem.CurrentSE.EntityType == proto.EntityDTO_VIRTUAL_MACHINE {
+		clusterAPIAction = true
+	}
 
 	// Acquire the lock for the actionItem. It blocks the action execution if the lock
 	// is used by other action. It results in error return if timed out (set in lockStore).
-	if lock, err := h.lockStore.getLock(actionItem); err != nil {
-		return err
-	} else {
-		// Unlock the entity after the action execution is finished
-		defer glog.V(4).Infof("Action %s: releasing lock", actionItem.GetUuid())
-		defer lock.ReleaseLock()
-		lock.KeepRenewLock()
-	}
-
-	// After getting the lock, need to get the k8s pod again as the previous action could delete the pod and create a new one.
-	// In such case, the action should be applied on the new pod.
-	// Currently, all actions need to get its related pod. If not needed, the pod is nil.
-	pod := h.getRelatedPod(actionItem)
-
-	if pod == nil {
-		err := fmt.Errorf("Cannot find the related pod for action item %s", actionItem.GetUuid())
-		return err
+	if !clusterAPIAction {
```

Comment:
Why do we ignore locks for machine related actions? I think it will be desirable to sequentialize the actions such that no concurrent actions can be applied to the same entity (machine). This may require us to update the `getLock` function to get the lock key for machine.

Reply:
We do that because we need to fail fast here, rather than wait. That was a requirement that came from Endre.